test/DynamoCoreWpfTests/ViewExtensions/PythonMigrationViewExtensionTests.cs

```diff
@@ -0,0 +1,235 @@
+﻿using Dynamo;
+using Dynamo.Configuration;
+using Dynamo.Interfaces;
+using Dynamo.Models;
+using Dynamo.Scheduler;
+using NUnit.Framework;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Windows;
+using Dynamo.PythonMigration;
+using System.Reflection;
+using System.Windows.Controls;
+using Dynamo.DocumentationBrowser;
+using Dynamo.Graph.Workspaces;
+using Dynamo.Wpf.Extensions;
+
+namespace DynamoCoreWpfTests
+{
+    class PythonMigrationViewExtensionTests : DynamoTestUIBase
+    {
+        private PythonMigrationViewExtension viewExtension = new PythonMigrationViewExtension();
+
+        private string PackagesDirectory { get { return Path.Combine(GetTestDirectory(this.ExecutingDirectory), "pkgs"); } }
+
+        protected override DynamoModel.IStartConfiguration CreateStartConfiguration(IPathResolver pathResolver)
+        {
+            return new DynamoModel.DefaultStartConfiguration()
+            {
+                PathResolver = pathResolver,
+                StartInTestMode = true,
+                GeometryFactoryPath = this.preloader.GeometryFactoryPath,
+                ProcessMode = TaskProcessMode.Synchronous,
+                Preferences = new PreferenceSettings() { CustomPackageFolders = new List<string>() { this.PackagesDirectory } }
+            };
+        }
+
+        /// <summary>
+        /// This test is created to check if the extension displays a dialog to the user
+        /// when opening a saved dyn file that contains python nodes with their engine set to `IronPython2`
+        /// </summary>
+        [Test]
+        public void WillDisplayDialogWhenOpeningGraphWithIronPythonNodes()
+        {
+            // Arrange
+            RaiseLoadedEvent(this.View);
+            var extensionManager = View.viewExtensionManager;
+            extensionManager.Add(viewExtension);
+
+            // Act
+            // open file
+            var examplePath = Path.Combine(UnitTestBase.TestDirectory, @"core\python", "python.dyn");
```

Comment:
suggest renaming the `.dyn` files to be a tiny bit more descriptive, something like `GraphWithIronPythonNode.dyn` for example 😛